cylc/flow/task_job_mgr.py

```diff
@@ -287,7 +287,13 @@ def submit_task_jobs(self, suite, itasks, curve_auth,
                 cmd.append('--debug')
             if get_utc_mode():
                 cmd.append('--utc-mode')
-            remote_mode = False
+            if is_remote_platform(itask.platform):
+                remote_mode = True
+                cmd.append('--remote-mode')
```

Comment:
This was the crucial missing line, because setting this causes files to be installed on the machine running the script.